config/static.c

```diff
@@ -135,6 +135,76 @@ l_gk_assign_lcores(lua_State *l)
 	return 0;
 }
 
+static int
+protected_gk_assign_sol_map(lua_State *l)
+{
+	uint32_t ctypeid;
+	struct gk_config *gk_conf;
+	lua_Integer i, n;
+	unsigned int *gk_sol_map;
+
+	gk_conf = *(struct gk_config **)
+		luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_GK_CONFIG_PTR);
+	n = lua_objlen(l, 2);
+	gk_sol_map = *(unsigned int **)lua_touserdata(l, 3);
+
+	for (i = 1; i <= n; i++) {
+		lua_pushinteger(l, i);	/* Push i. */
+		lua_gettable(l, 2);	/* Pop i, Push t[i]. */
+
+		/* Check that t[i] is a number. */
+		if (!lua_isnumber(l, -1))
+			luaL_error(l, "Index %i is not a number", i);
+		gk_sol_map[i - 1] = lua_tointeger(l, -1) - 1;
+
+		lua_pop(l, 1);		/* Pop t[i]. */
+	}
+
+	gk_conf->gk_sol_map = gk_sol_map;
+	return 0; /* No results. */
+}
+
+static int
+l_gk_assign_sol_map(lua_State *l)
+{
+	uint32_t ctypeid;
+	lua_Integer n;
+	unsigned int *gk_sol_map, **ud;
+	uint32_t correct_ctypeid = luaL_get_ctypeid(l,
+		CTYPE_STRUCT_GK_CONFIG_PTR);
+
+	/* First argument must be of type CTYPE_STRUCT_GK_CONFIG_PTR. */
+	luaL_checkcdata(l, 1, &ctypeid, CTYPE_STRUCT_GK_CONFIG_PTR);
+	if (ctypeid != correct_ctypeid)
+		luaL_error(l, "Expected `%s' as first argument",
+			CTYPE_STRUCT_GK_CONFIG_PTR);
+
+	/* Second argument must be a table. */
+	luaL_checktype(l, 2, LUA_TTABLE);
+
+	n = lua_objlen(l, 2); /* Get size of the table. */
+	if (n <= 0)
+		return 0; /* No results. */
+
+	ud = lua_newuserdata(l, sizeof(gk_sol_map));
+
+	lua_pushcfunction(l, protected_gk_assign_sol_map);
+	lua_insert(l, 1);
+
+	gk_sol_map = rte_calloc("gk_conf.gk_sol_map", n,
```

Comment:
Why `rte_calloc()` instead of `rte_malloc()` as used in similar functions? `gk_sol_map` is only returned at the end of `protected_gk_assign_sol_map()` when it was fully filled in and all went well.